kpi/tests/api/v2/test_api_submissions.py

```diff
@@ -215,6 +215,20 @@ def test_list_submissions_anonymous_asset_publicly_shared(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.asset.remove_perm(anonymous_user, 'view_submissions')
 
+    def test_list_submissions_authenticated_asset_publicly_shared(self):
+        # https://github.com/kobotoolbox/kpi/issues/2698
+        self._log_in_as_another_user()
+
+        # give the poor schmuck their own asset; needed to expose flaw in
+        # `ObjectPermissionMixin.__get_object_permissions()`
+        Asset.objects.create(name='i own it', owner=self.anotheruser)
+
+        anonymous_user = get_anonymous_user()
+        self.asset.assign_perm(anonymous_user, 'view_submissions')
```

Comment:
yep, literally copied and pasted from your `test_list_submissions_anonymous_asset_publicly_shared()` :wink:
maybe this isn't the place for a bigger discussion, but what's the point of `PERM_VIEW_SUBMISSIONS` anyway? should something like `_submitted_by` be replaced globally with `FIELD_SUBMITTED_BY`?
https://github.com/kobotoolbox/kpi/blob/06962e7329d8ceda672549ce971d951ee4301fb5/kpi/tests/api/v2/test_api_submissions.py#L203

i'm sure there's a good reason for it, but i don't know what it is :smile:

Reply:
`PERM_*` constants are used (almost) everywhere instead of their respective values. 
I think that's me who changed this when coding row level permission because one of the value changed quite a bit. IMHO, it's also easier to use the constant than a string. 
If we want to refactor the name of the permission, we just have to change the constant value.  

Because `_submitted_by` is fixed, I would say yes. 

I have to admit it doesn't bring other advantage other than that (if an advantage it is).

Reply:
OK, I understand this:
> If we want to refactor the name of the permission, we just have to change the constant value.

It does seem like it would be weird if, say, we decided to change `view_submissions` to `view_data` and did it by setting `PERM_VIEW_SUBMISSIONS = 'view_data'`, but I'm sure there are cases where that would make sense. I guess some constant names are easier to search for than their string values, too, although these aren't good examples of that.

Reply:
I'll change this PR to use `PERM_VIEW_SUBMISSIONS` since that's already the lay of the land. We can discuss further constant-ifying in the future :)

Reply:
Done